lib/text/ttml_text_parser.js

```diff
@@ -360,22 +360,24 @@ shaka.text.TtmlTextParser = class {
     if (extent) {
       percentage = TtmlTextParser.percentValues_.exec(extent);
       results = percentage || TtmlTextParser.pixelValues_.exec(extent);
-      if (results != null) {
-        if (globalWidth != null) {
-          region.width = Number(results[1]) * 100 / globalWidth;
-        } else {
-          region.width = Number(results[1]);
-        }
-        if (globalHeight != null) {
-          region.height = Number(results[2]) * 100 / globalHeight;
-        } else {
-          region.height = Number(results[2]);
+      if (results !== null) {
```

Comment:
Because our code is compiled and uses type info during compilation, triple-equals is rarely needed in this codebase, and may even add unhelpful strictness at runtime.

For example, imagine someone installs a polyfill or shim for some browser-builtin, and that polyfill isn't quite perfect.  Imagine `exec()` above returns `undefined` instead of `null`.  `undefined == null`, but `undefined !== null`, so we would then try to extract a number from `undefined`.

For first-party code that goes through the compiler, types are checked at compile time, so if a variable gets assigned an unexpected type in our source, compilation fails.  That is almost always sufficient.

With double-equals, null is equal to undefined, but not to 0, "", or false.  There's really no reason to use triple-equals on null checks.

Reply:
thanks for the insight, resolved with a fresh commit